### PR TITLE
[MIG][17.0] document_page_approval

### PR DESCRIPTION
**Pull request by EdgarRetes**
@max3903

### CODE REVIEW

**Comment by max3903** [comment]
@EdgarRetes I get an error message when editing a controlled category:
```AttributeError: 'document.page.history' object has no attribute 'message_post_with_template'```

File: document_page_approval/models/document_page.py [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+
+    def _create_history(self, vals):
+        res = super()._create_history(vals)
+        res.action_to_approve()
+        return res
+
+    def action_changes_pending_approval(self):
+        self.ensure_one()
+        action = self.env.ref("document_page_approval.action_change_requests")
+        action = action.sudo().read()[0]
```
Comment (victoralmau) [inline document_page_approval/models/document_page.py]:
```suggestion
        action = self.env["ir.actions.act_window"]._for_xml_id(
            "document_page_approval.action_change_requests"
        )
```

File: document_page_approval/security/ir.model.access.csv [hunk 2]
```diff
@@ -0,0 +1 @@
+id,name,model_id:id,group_id:id,perm_read,perm_write,perm_create,perm_unlink
```
Comment (victoralmau) [inline document_page_approval/security/ir.model.access.csv:1]: This empty file makes no sense, I think the best thing to do is to delete it.

File: document_page_approval/views/document_page_approval.xml [hunk 3]
```diff
@@ -0,0 +1,305 @@
...
+        <field name="arch" type="xml">
+            <sheet position="before">
+                <header>
+                    <!-- draft -> to approve -->
+                    <button
+                        name="action_to_approve"
+                        type="object"
+                        string="Send to Review"
+                        class="oe_highlight"
+                        invisible="not is_approval_required or not am_i_owner or state not in 'draft'"
```
Comment (victoralmau) [inline document_page_approval/views/document_page_approval.xml]:
```suggestion
                        invisible="not is_approval_required or not am_i_owner or state != 'draft'"
```

File: document_page_approval/views/document_page_approval.xml [hunk 4]
```diff
@@ -0,0 +1,305 @@
...
+                        type="object"
+                        string="Cancel"
+                        invisible="(not am_i_owner or not am_i_approver) and (state not in ('draft', 'to approve'))"
+                    />
+                    <!-- reopen if i am owner or approver -->
+                    <button
+                        name="action_draft"
+                        type="object"
+                        string="Back to draft"
+                        invisible="(not am_i_owner or not am_i_approver) and (state not in 'cancelled')"
```
Comment (victoralmau) [inline document_page_approval/views/document_page_approval.xml]:
```suggestion
                        invisible="(not am_i_owner or not am_i_approver) and (state != 'cancelled')"
```

File: document_page_approval/views/document_page_approval.xml [hunk 5]
```diff
@@ -0,0 +1,305 @@
...
+                    />
+                    <!-- cancel & reopen, if i am owner or approver -->
+                    <button
+                        name="action_draft"
+                        type="object"
+                        string="Back to draft"
+                        invisible="(not am_i_owner or not am_i_approver) and (state not in 'to approve')"
```
Comment (victoralmau) [inline document_page_approval/views/document_page_approval.xml]:
```suggestion
                        invisible="(not am_i_owner or not am_i_approver) and (state != 'to approve')"
```

**Review by victoralmau — CHANGES_REQUESTED** [review]
Code review:

- The document_page changes must be correct and not just comment code.
- Can you update the example images (they are quite old)?
- All changes must be in the Migration to 17.0 commit
- Can you squahs the administrative commits?

File: document_page_approval/security/ir.model.access.csv (hunk 2 above)
Comment (EdgarRetes) [reply document_page_approval/security/ir.model.access.csv:1]: I think this file is necessary for the tests

File: document_page/models/document_page_history.py [hunk 6]
```diff
@@ -66,11 +66,5 @@ def _get_diff(self, v1, v2):
                 context=True,
             )
 
-    # TODO: Replace for _compute_display_name
     def name_get(self):
```
Comment (victoralmau) [inline document_page/models/document_page_history.py:69]: Remove the `name_get()` method, but do NOT remove the `_compute_display_name()` method. That's what is stated in the migration guidelines https://github.com/OCA/maintainer-tools/wiki/Migration-to-version-17.0

File: document_page_approval/security/ir.model.access.csv (hunk 2 above)
Comment (victoralmau) [reply document_page_approval/security/ir.model.access.csv:1]: It is not necessary and does nothing, remove it.

**Review by victoralmau — CHANGES_REQUESTED** [review]

**Comment by victoralmau** [comment]
Superseded by https://github.com/OCA/knowledge/pull/509

**Pull request closed without merging**